MantidQt/CustomInterfaces/src/Reflectometry/ReflMainViewPresenter.cpp

```diff
@@ -63,10 +63,13 @@ ReflMainViewPresenter::ReflMainViewPresenter(
   const std::string defaultInst =
       Mantid::Kernel::ConfigService::Instance().getString("default.instrument");
   if (std::find(instruments.begin(), instruments.end(), defaultInst) !=
-      instruments.end())
+      instruments.end()) {
     m_view->setInstrumentList(instruments, defaultInst);
-  else
+    m_tablePresenter->setInstrumentList(instruments, defaultInst);
+  } else {
     m_view->setInstrumentList(instruments, "INTER");
+    m_tablePresenter->setInstrumentList(instruments, "INTER");
```

Comment:
Also good. if you change this to a boost::weak_ptr. I suggest using the shared_ptr construct to guarantee that the code throws if the m_tablePresenter has somehow been deleted.